backend/corpora/common/corpora_orm.py

```diff
@@ -37,40 +39,61 @@ class TransformingBase(object):
+    ) -> dict:
         """
         Converts the columns and relationships of a SQLAlchemy Base object into a python dictionary.
 
         :param backref: used to avoid recursively looping between two tables.
         :param remove_none: If true, removes keys that are none from the result.
+        :param remove_attr: Attributes not to convert.
+        :param remove_relationships: Ignore relationships.
         :return: a dictionary representation of the database object.
         """
+        result = dict()
+        remove_attr = remove_attr if remove_attr else []
 
         # Populate result with columns.
-        if remove_none:
-            result = {
-                column.key: getattr(self, attr)
-                for attr, column in self.__mapper__.c.items()
-                if getattr(self, attr) is not None
-            }
-        else:
-            result = {column.key: getattr(self, attr) for attr, column in self.__mapper__.c.items()}
+        for attr, column in self.__mapper__.c.items():
+            if column.key in remove_attr:
+                continue
+            if remove_none:
+                if getattr(self, attr) is not None:
+                    result[column.key] = getattr(self, attr)
+                else:
+                    continue
+            else:
+                result[column.key] = getattr(self, attr)
 
         # Populate result with relationships.
-        for attr, relation in self.__mapper__.relationships.items():
-            # Avoid recursive loop between two tables.
-            if backref == relation.target:
-                continue
-            value = getattr(self, attr)
-            if value is None:
-                if not remove_none:
-                    result[relation.key] = None
-            elif isinstance(value.__class__, DeclarativeMeta):
-                result[relation.key] = value.to_dict(self.__table__, remove_none)
-            elif isinstance(value, list):
-                result[relation.key] = [i.to_dict(self.__table__, remove_none) for i in value]
+        if not remove_relationships:
+            if backref:
+                backref.append(self.__table__)
             else:
-                raise CorporaException(f"Unable to convert to dictionary. Unexpected type: {type(value)}.")
+                backref = [self.__table__]
+
+            for attr, relation in self.__mapper__.relationships.items():
+                if attr in remove_attr:
+                    continue
+                # Avoid recursive loop between multiple tables.
+                if relation.target in backref:
+                    continue
+                value = getattr(self, attr)
+                if value is None:
+                    if not remove_none:
+                        result[relation.key] = None
+                elif isinstance(value.__class__, DeclarativeMeta):
+                    result[relation.key] = value.to_dict(backref=backref, remove_none=remove_none)
```

Comment:
would we ever want to pass through the `remove_attr`?

Reply:
There are not uses cases for us that I can think of. It's easy enough to add in later if we want.